snapcraft/cli/_channel_map.py

```diff
@@ -123,20 +123,28 @@ def _get_channel_lines_for_channel(  # noqa: C901
             progressive_mapped_channel.revision
         )
 
+        if progressive_mapped_channel.progressive.percentage is None:
```

Comment:
this should probably raise a RuntimeError as "get_mapped_channel" with progressive True will return a channel where percentage is set

Reply:
Makes sense! Updated.